pkg/datagatherer/aks/aks.go

```diff
@@ -0,0 +1,107 @@
+// Package aks provides a datagatherer for AKS.
+package aks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	aks "github.com/Azure/aks-engine/pkg/api/agentPoolOnlyApi/v20180331"
+)
+
+// AKSDataGatherer is a DataGatherer for AKS.
+type AKSDataGatherer struct {
+	ctx           context.Context
+	resourceGroup string
+	clusterName   string
+	credentials   *AzureCredentials
+}
+
+// AKSInfo contains the data retrieved from AKS.
+type AKSInfo struct {
+	Cluster *aks.ManagedCluster
+}
+
+type AzureCredentials struct {
+	AccessToken  string `json:"accessToken"`
+	ExpiresOn    string `json:"expiresOn"`
+	Subscription string `json:"subscription"`
+	Tenant       string `json:"tenant"`
+	TokenType    string `json:"tokenType"`
+}
+
+func readCredentials(path string) (*AzureCredentials, error) {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var creds AzureCredentials
+	err = json.Unmarshal(b, &creds)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(creds.Subscription) == 0 {
+		return nil, fmt.Errorf("'subscription' must not be empty")
+	}
+	if creds.TokenType != "Bearer" {
+		return nil, fmt.Errorf("'tokenType' %s is not supported", creds.TokenType)
+	}
+
+	return &creds, nil
+}
+
+// NewAKSDataGatherer creates a new AKSDataGatherer for a cluster.
+func NewAKSDataGatherer(ctx context.Context, resourceGroup, clusterName, credentialsPath string) (*AKSDataGatherer, error) {
+	credentials, err := readCredentials(credentialsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return &AKSDataGatherer{
+		ctx:           ctx,
+		resourceGroup: resourceGroup,
+		clusterName:   clusterName,
+		credentials:   credentials,
+	}, nil
+}
+
+// Fetch retrieves cluster information from AKS.
+func (g *AKSDataGatherer) Fetch() (interface{}, error) {
+	client := &http.Client{}
```

Comment:
Is there a reason for using the HTTP client rather than the Azure client? https://github.com/Azure-Samples/azure-sdk-for-go-samples/blob/master/compute/container_cluster.go

Reply:
I don't have a strong opinion about how it should be done, just comparing it to other gatherers.

Reply:
Actually, I didn't know about https://github.com/Azure-Samples/azure-sdk-for-go-samples.

I checked https://github.com/Azure/azure-sdk-for-go and https://github.com/Azure/aks-engine and none of them can serve as a client for what we wanted here.

I will give https://github.com/Azure-Samples/azure-sdk-for-go-samples a try.